lib/compat/experimental/persisted-preferences.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Server-side requirements for database persisted preferences.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Register the user meta for persisted preferences.
+ */
+function gutenberg_configure_persisted_preferences() {
+	$user_id = get_current_user_id();
+	if ( empty( $user_id ) ) {
+		return;
+	}
+
+	global $wpdb;
+
+	$meta_key = $wpdb->get_blog_prefix() . 'persisted_preferences';
+	register_meta(
+		'user',
+		$meta_key,
+		array(
+			'type'         => 'object',
+			'single'       => true,
+			'show_in_rest' => array(
+				'name'   => 'persisted_preferences',
+				'type'   => 'object',
+				'schema' => array(
+					'type'                 => 'object',
+					'properties'           => array(),
+					'additionalProperties' => true,
+				),
+			),
+		)
+	);
+
+	$preload_data = get_user_meta( $user_id, $meta_key, true );
+
+	wp_add_inline_script(
+		'wp-database-persistence-layer',
```

Comment:
I don't understand why you chose to add this to `wp-database-persistence-layer` instead of `wp-preferences` (which mean you don't have to add any dependencies to any script as `wp-preferences`)

Reply:
Hmm, true. I'm still learning about the best way to do this. Thanks for the pointers.